test/parallel/proofOfStake.3rdPartyRewards.activation.check.test.ts

```diff
@@ -0,0 +1,378 @@
+      await Sudo.batchAsSudoFinalized(
+        Assets.promotePool(liqId.toNumber(), 20),
+        Assets.mintToken(liqId, testUser, Assets.DEFAULT_AMOUNT),
+      );
+
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+        expect(res.data).toEqual("NotAPromotedPool");
+      });
+    });
+
+    test("GIVEN promoted pool MGX-Token2, pool MGX-Token1 AND user without liquidity tokens for MGX-Token2 WHEN  user tries to activate 3rd party rewards for Token1 THEN receive error", async () => {
+      await createAndPromoteSecondPool(BN_ZERO, true);
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+        expect(res.data).toEqual("NotEnoughAssets");
+      });
+    });
+
+    test("GIVEN promoted pool MGX-Token2, pool MGX-Token1 AND user with liquidity tokens for MGX-Token2 WHEN user tries to activate 3rd party rewards for Token1 with too many liquidity tokens THEN receive error", async () => {
+      await createAndPromoteSecondPool(Assets.DEFAULT_AMOUNT, true);
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT.muln(2),
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+        expect(res.data).toEqual("NotEnoughAssets");
+      });
+    });
+
+    test("GIVEN promoted pool MGX-Token2, pool MGX-Token1 AND user with liquidity tokens for MGX-Token2 WHEN user activates 3rd party rewards for Token1 AND does not uses rewardPool function THEN he cant claim rewards", async () => {
+      await createAndPromoteSecondPool(Assets.DEFAULT_AMOUNT.muln(2), false);
+      await signTx(
+        getApi(),
+        await ProofOfStake.activateLiquidityFor3rdpartyRewards(
+          liqId,
+          Assets.DEFAULT_AMOUNT,
+          newToken,
+        ),
+        testUser.keyRingPair,
+      ).then((events) => {
+        const res = getEventResultFromMangataTx(events);
+        expect(res.state).toEqual(ExtrinsicResult.ExtrinsicFailed);
+        expect(res.data).toEqual("NotAPromotedPool");
+      });
```

Comment:
this test is about claiming too, plz ensure that claiming fails too.

Reply:
Done